### PR TITLE
Added Spring Readme, Updated Java & Micronaut

### PR DESCRIPTION
**Pull request by sachin-pikle**
No code changes. Only added and updated readme files.

New readme files added to run the Spring Native example in OCI Code Editor and OCI Cloud Shell.
Moved the software version and environment variables checks to a common folder instead of duplicating the same in multiple samples.
Updated `java-hello-world-maven` and `micronaut-hello-rest-maven` readme files.

### CODE REVIEW

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 1]
```diff
@@ -147,15 +166,27 @@ This step is optional - [Check software version and environment variables](./REA
...
+3. Run the native executable using:
+
+    ```shell
+    ./target/my-app
+    ```
+
+    The output should be similar to:
+    ```
+    Hello World!
+    ```
```
Comment (olyagpl) [inline java-hello-world-maven/README-Cloud-Shell.md:178]: Why do you leave empty lines?

**Review by olyagpl — APPROVED** [review]

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 1 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md:178]: Empty lines is for better readability. Whitespace is easier on the eyes. :-)

File: _common/README-check-version-env-vars.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-## Step 3: [OPTIONAL] Confirm Software Version and Environment Variables
+## [OPTIONAL] Confirm software version and environment variables
```
Comment (fniephaus) [inline _common/README-check-version-env-vars.md:1]: Do we really need a new `_common` directory for a single file?

File: _common/README-check-version-env-vars.md [hunk 3]
```diff
@@ -78,4 +78,4 @@
     ...
     ```
 
-Continue to **[Step 4: Setup Project and Run](./README-CS.md#step-4-setup-project-and-run)**
\ No newline at end of file
+Go back to continue with the rest of the demo sample.
```
Comment (bhoran) [inline _common/README-check-version-env-vars.md:81]:
```suggestion
Go back to continue with the rest of the demo.
```

File: _common/README-check-version-env-vars.md (hunk 2 above)
Comment (sachin-pikle) [reply _common/README-check-version-env-vars.md:1]: Yes, we have three to four samples using this file at the moment and more to come. If there are other common files in the future, we can add them to this folder.

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 4]
```diff
@@ -4,17 +4,15 @@ This sample shows how you can get started quickly with GraalVM Enterprise Editio
...
 
 GraalVM Enterprise Edition is available for use on Oracle Cloud Infrastructure (OCI) at no additional cost.
 
 ## What is Cloud Shell?
 
-Cloud Shell is a free-to-use browser-based terminal accessible from the Oracle Cloud Console. It provides access to a Linux shell with pre-authenticated OCI CLI and other pre-installed developer tools. You can use the shell to interact with OCI resources, follow labs and tutorials, and quickly run commands. 
+[Cloud Shell](https://www.oracle.com/devops/cloud-shell/) is a free-to-use browser-based terminal accessible from the Oracle Cloud Console. It provides access to a Linux shell with preinstalled developer tools and a pre-authenticated OCI CLI. You can use the shell to interact with OCI resources, follow labs and tutorials, and quickly run utility commands.
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:13]:
```suggestion
[Cloud Shell](https://www.oracle.com/devops/cloud-shell/) is a free-to-use browser-based terminal accessible from the Oracle Cloud Console. It provides access to a Linux shell with preinstalled developer tools and a preauthenticated OCI CLI. You can use the shell to interact with OCI resources, follow labs and tutorials, and quickly run utility commands.
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 5]
```diff
@@ -48,14 +46,31 @@ GraalVM Enterprise Native Image and JDK 17 are preinstalled in Cloud Shell.
     The current managed java version is set to graalvmeejdk-17.0.4.
     ```
 
-## Step 3: [OPTIONAL] Confirm Software Version and Environment Variables
+## Step 3: [OPTIONAL] Confirm software version and environment variables
+
+This step is optional - [Check software version and environment variables](../_common/README-check-version-env-vars.md)
+
+
+## Step 4: Setup project, build and run as a JAR
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:54]:
```suggestion
## Step 4: Set up your project, build and run as a JAR
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 6]
```diff
@@ -48,14 +46,31 @@ GraalVM Enterprise Native Image and JDK 17 are preinstalled in Cloud Shell.
...
+
+1. Git clone this sample folder.
+
+    ```shell
+    git init graalvmee-java-hello-world-maven
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:59]: Why "init" and not "clone"?

Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md:59]: The group of commands together clones just a single sample folder, not the entire graalvm-demos repo.

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 7]
```diff
@@ -48,14 +46,31 @@ GraalVM Enterprise Native Image and JDK 17 are preinstalled in Cloud Shell.
...
-1. Git clone this repo.
+    echo "java-hello-world-maven/*">>.git/info/sparse-checkout
+
+    git pull --depth=1 origin master
+
+    cd java-hello-world-maven
+
+    ```
 
 2. Build a JAR for the sample app.
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:75]:
```suggestion
2. Build a JAR file for the example/demo app.
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 8]
```diff
@@ -74,17 +89,21 @@ This step is optional - [Check software version and environment variables](./REA
     Hello World!
     ```
 
-4. Let's use GraalVM Native Image to produce a native executable.
+## Step 5: Build and run as a native executable
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:92]:
```suggestion
## Step 5: Build and run a native executable
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 9]
```diff
@@ -74,17 +89,21 @@ This step is optional - [Check software version and environment variables](./REA
...
+
+Let's use GraalVM Native Image to produce a native executable.
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:94]:
```suggestion
Use GraalVM Native Image to produce a native executable.
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 10]
```diff
@@ -74,17 +89,21 @@ This step is optional - [Check software version and environment variables](./REA
...
+
+**Option 1: Quick Build enabled**
 
-    4.1) **Option 1: Quick Build enabled**
+You will notice the `Quick Build` mode reduces the time required to generate a native executable, making it covenient to use Native Image builds in a typical development cycle (edit, compile, test, and debug). The `Quick Build` mode is recommended for development purpose only.
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:98]:
```suggestion
You will notice the `Quick Build` mode reduces the time required to generate a native executable, making it covenient to use Native Image builds in a typical development cycle (edit, compile, test, and debug). The `Quick Build` mode is recommended for development purposes only.
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 11]
```diff
@@ -74,17 +89,21 @@ This step is optional - [Check software version and environment variables](./REA
...
 
-    To enable `Quick Build`, open [pom.xml](./pom.xml) in a text editor like Nano and uncomment the line shown:
+1. To enable `Quick Build`, open [pom.xml](./pom.xml) in a text editor like Nano and uncomment the line shown:
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:100]:
```suggestion
1. To enable `Quick Build`, open [pom.xml](./pom.xml) in a text editor such as Nano and uncomment the line shown:
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 12]
```diff
@@ -74,17 +89,21 @@ This step is optional - [Check software version and environment variables](./REA
...
 
     ```
     <buildArg>-Ob</buildArg>
     ```
 
-    Run the Native Image build to generate a native executable:
+2. Run the Native Image maven build to generate a native executable:
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:106]:
```suggestion
2. Run the Native Image maven build to generate a native executable:
```
This sentence is a bit clumsy. You could try "Use the Native Image maven plugin to  create a native executable"

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 13]
```diff
@@ -147,15 +166,27 @@ This step is optional - [Check software version and environment variables](./REA
...
+    The output should be similar to:
+    ```
+    Hello World!
+    ```
+
+
+**Option 2: Quick Build disabled** 
     
-    To disable `Quick Build`, open [pom.xml](pom.xml) in a text editor like Nano and comment the line shown:  
+1. To disable `Quick Build`, open [pom.xml](pom.xml) in a text editor like Nano and comment the line shown:
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:183]:
```suggestion
1. To disable `Quick Build`, open [pom.xml](pom.xml) in a text editor such as Nano and comment the line shown:  
```

File: java-hello-world-maven/README-Cloud-Shell.md [hunk 14]
```diff
@@ -147,15 +166,27 @@ This step is optional - [Check software version and environment variables](./REA
...
     
-    To disable `Quick Build`, open [pom.xml](pom.xml) in a text editor like Nano and comment the line shown:  
+1. To disable `Quick Build`, open [pom.xml](pom.xml) in a text editor like Nano and comment the line shown:  
 
     ```
     <!-- <buildArg>-Ob</buildArg> -->
     ```
 
-    Run the Native Image build to generate a native executable:
+2. Run the Native Image maven build to generate a native executable:
```
Comment (bhoran) [inline java-hello-world-maven/README-Cloud-Shell.md:189]: See above.

File: micronaut-hello-rest-maven/README-Cloud-Shell.md [hunk 15]
```diff
@@ -0,0 +1,224 @@
+# Micronaut Hello World REST App with GraalVM Enterprise in OCI Cloud Shell
+
+This sample shows how you can get started quickly with GraalVM Enterprise Edition in Oracle Cloud Infrastructre (OCI) Cloud Shell. This sample uses a simple hello world REST application built with the Micronaut framework and GraalVM Enterprise Native Image and JDK (Java Development Kit).
```
Comment (bhoran) [inline micronaut-hello-rest-maven/README-Cloud-Shell.md:3]: I think I prefer "example", but that's a personal preference. It's probably more commonly used than  "sample".

File: spring-native-image/README.md [hunk 16]
```diff
@@ -1,24 +1,29 @@
...
+
+- [GraalVM Enterprise in OCI Code Editor](./README-Code-Editor.md)
+- [GraalVM Enterprise in OCI Cloud Shell](./README-Cloud-Shell.md)
+- On other machines with Linux or Mac or Windows, follow the instuctions given below.
+
+## Other machines with Linux or Mac or Windows
+
+### Note on building Docker images with native executables inside on MAC
 
 If you are using a Mac or Windows in order to build a Docker image containing your native executable you will need
```
Comment (bhoran) [inline spring-native-image/README.md:23]:
```suggestion
If you are using macOS or Windows in order to build a Docker image containing your native executable you will need
```

File: spring-native-image/README.md [hunk 17]
```diff
@@ -49,7 +54,7 @@ docker run --rm --name graalce -d -p 8080:8080 jibber-benchmark:graalee.0.0.1-SN
 You can then test the container suing `curl` exactly as you did before - remember to allow a little time for the application to 
 start up.
 
-## Building & Running the GraalVm Native Image Executable
+### Build and run as a native executable
```
Comment (bhoran) [inline spring-native-image/README.md:57]:
```suggestion
### Build and run a native executable
```

File: spring-native-image/README.md [hunk 18]
```diff
@@ -75,7 +80,7 @@ fg
 
 From the log output that the executable outputs, you can see that it starts much faster than the Java version.
 
-## Containerising the Native Executable
+### Containerising the Native Executable
```
Comment (bhoran) [inline spring-native-image/README.md:83]:
```suggestion
### Containerizing the Native Executable
```

File: _common/README-check-version-env-vars.md (hunk 3 above)
Comment (sachin-pikle) [reply _common/README-check-version-env-vars.md]: Updated to use `example` in all the places.

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 4 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Changed to `preauthenticated`

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 5 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 7 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Used `example`

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 8 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done.

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 9 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done.

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 10 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 11 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 12 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done.

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 13 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done

File: java-hello-world-maven/README-Cloud-Shell.md (hunk 14 above)
Comment (sachin-pikle) [reply java-hello-world-maven/README-Cloud-Shell.md]: Done

File: micronaut-hello-rest-maven/README-Cloud-Shell.md (hunk 15 above)
Comment (sachin-pikle) [reply micronaut-hello-rest-maven/README-Cloud-Shell.md]: Changed to `example` in all the readme files.

File: spring-native-image/README.md (hunk 16 above)
Comment (sachin-pikle) [reply spring-native-image/README.md]: Done.

File: spring-native-image/README.md (hunk 17 above)
Comment (sachin-pikle) [reply spring-native-image/README.md]: Done.

File: spring-native-image/README.md (hunk 18 above)
Comment (sachin-pikle) [reply spring-native-image/README.md]: Done

**Review by olyagpl — APPROVED** [review]

**Pull request merged**